contrib/sentence_transformers/test_sentence_transformers.py

```diff
@@ -0,0 +1,34 @@
+import pytest
+
+from sentence_transformers import SentenceTransformer, util
+
+from ..utils import production_endpoint
+
+
+@pytest.fixture(scope="module")
+def multi_qa_model() -> SentenceTransformer:
+    with production_endpoint():
+        return SentenceTransformer("multi-qa-MiniLM-L6-cos-v1")
+
+
+def test_from_pretrained(multi_qa_model: SentenceTransformer) -> None:
+    # Example taken from https://www.sbert.net/docs/hugging_face.html#using-hugging-face-models.
+    query_embedding = multi_qa_model.encode("How big is London")
+    passage_embedding = multi_qa_model.encode(
+        [
+            "London has 9,787,426 inhabitants at the 2011 census",
+            "London is known for its financial district",
+        ]
+    )
+    print("Similarity:", util.dot_score(query_embedding, passage_embedding))
+
+
+@pytest.mark.xfail(
+    reason=(
+        "Production endpoint is hardcoded in sentence_transformers when pushing to Hub."
+    )
+)
+def test_push_to_hub(
+    multi_qa_model: SentenceTransformer, repo_name: str, cleanup_repo: None
+) -> None:
+    multi_qa_model.save_to_hub(repo_name)
```

Comment:
Would be nice to eventually have a test that doesn't fail to ensure that save to hub actually works :) we could have a specific org for that, like skops does, but I understand it's a bit complex to setup + very annoying to have testing artifacts on the actual hub

Reply:
(discussed offline) Decision has been taken that the `contrib/` test suite purpose is only to test the deprecation warnings in downstream libraries. Testing the validity of a pushed/downloaded model is therefore out of scope here.
This can be reevaluated in the future :)

Reply:
> Would be nice to eventually have a test that doesn't fail to ensure that save to hub actually works :)

Yes, completely agree on that. I'd like to do that later. I am about to open an issue/PR on sentence transformers side to test that properly.
Worse case scenario, I have set a reminder to myself in 10 days.